source/base/mpi_compute_index_owner_internal.cc

```diff
@@ -638,7 +639,7 @@ namespace Utilities
                   send_requests.push_back(MPI_Request());
                   const int ierr = MPI_Isend(send_data[i].data(),
                                              send_data[i].size(),
-                                             MPI_UNSIGNED,
+                                             MPI_UNSIGNED_LONG,
```

Comment:
Can you please adjust this type such that it both works with 32 and 64 bit integers (note the type of `types::global_dof_index`?

Reply:
Done!